api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const accessToken = req.get('X-Access-Token');
+  if (!accessToken) {
+    throw new ClientError(401, 'error: authentication required');
+  }
+  try {
+    const payload = jwt.verify(accessToken, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
Be sure to pass the `err` to `next()` so that the client receives a response and doesn't hang